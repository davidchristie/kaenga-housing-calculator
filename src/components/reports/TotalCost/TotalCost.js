/* eslint-disable react/style-prop-object */

import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Well } from 'react-bootstrap'

import Currency from '../../units/Currency'

class TotalCost extends Component {
  render () {
    const {
      buildCost,
      landCost,
      total
    } = this.props
    return (
      <Well>
        <h1>Total Cost</h1>
        <p>
          Build cost: <strong><Currency value={buildCost} /></strong>
        </p>
        <p>
          Land cost: <strong><Currency value={landCost} /></strong>
        </p>
        <br />
        <p>
          Total: <strong><Currency value={total} /></strong>
        </p>
      </Well>
    )
  }
}

TotalCost.defaultProps = {
  buildCost: 0,
  total: 0
}

TotalCost.propTypes = {
  buildCost: PropTypes.number,
  total: PropTypes.number
}

export default TotalCost
