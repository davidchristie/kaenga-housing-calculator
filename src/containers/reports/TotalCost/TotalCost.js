import { connect } from 'react-redux'

import Report from '../../../components/reports/TotalCost'
import { getReport } from '../../../selectors/report'

export default connect(
  state => {
    return {
      ...getReport(state)
    }
  }
)(Report)
