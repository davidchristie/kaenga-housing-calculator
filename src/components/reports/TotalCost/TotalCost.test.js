import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import TotalCost from './TotalCost'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<TotalCost />)).toMatchSnapshot()
})
