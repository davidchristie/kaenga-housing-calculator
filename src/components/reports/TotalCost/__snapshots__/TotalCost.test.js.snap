exports[`test matches snapshot 1`] = `
<Well
  bsClass="well">
  <h1>
    Total Cost
  </h1>
  <p>
    Build cost: 
    <strong>
      <Currency
        type="nzd"
        value={0} />
    </strong>
  </p>
  <p>
    Land cost: 
    <strong>
      <Currency
        type="nzd"
        value={0} />
    </strong>
  </p>
  <br />
  <p>
    Total: 
    <strong>
      <Currency
        type="nzd"
        value={0} />
    </strong>
  </p>
</Well>
`;
